### PR TITLE
Use a unique hashing function when encoding sum_fn's equality

### PR DESCRIPTION
**Pull request by makesource**
Based on previous analysis, we can verify all cases by using a unique hashing function.

### CODE REVIEW

File: src/abstractops.cpp [hunk 1]
```diff
@@ -777,7 +777,7 @@ Expr AbsFpEncoding::getFpAssociativePrecondition() const {
       if (!an.isUInt(alen) || !bn.isUInt(blen) || alen != blen) continue;
 
       auto domainSort = a.select(Index(0)).sort();
-      FnDecl hashfn(domainSort, Index::sort(), freshName("fp_hash"));
+      FnDecl hashfn(domainSort, Index::sort(), "fp_hash");
```
Comment (aqjune) [inline src/abstractops.cpp:780]: Should we change the range (치역) of fp_hash?

Comment (makesource) [reply src/abstractops.cpp:780]: The range is already Index::sort() (which is BV(32))
Should it be more optimized or determined dynamically?

**Comment by aqjune** [comment]
I think so. Tensor size is guaranteed to be within Index::sort, but not m^2
n is.

2021년 12월 4일 (토) 오후 11:51, seongwon bang ***@***.***>님이 작성:

File: src/abstractops.cpp (hunk 1 above)
Comment (aqjune) [reply src/abstractops.cpp:780]: Another comment: You'll want to declare FnDecl only once & reuse it. (e.g. AbsFpEncoding::getAddFn)

**Pull request merged**